include/matx/operators/legendre.h

```diff
@@ -0,0 +1,170 @@
+//    list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice,
+//    this list of conditions and the following disclaimer in the documentation
+//    and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its
+//    contributors may be used to endorse or promote products derived from
+//    this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+/////////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+
+#include "matx/core/type_utils.h"
+#include "matx/operators/base_operator.h"
+
+namespace matx
+{
+  /**
+   * Legendre polynomial
+   *
+   * Calculates the Nth order legrande polynomial evaluated
+   * at the input X
+   */
+  namespace detail {
+    template <typename T1>
+      class LegendreOp : public BaseOp<LegendreOp<T1>>
+    {
+      private:
+        typename base_type<T1>::type in_;
+        int order_;
+        int axis_;
+
+        template<class TypeParam>
+          static __MATX_INLINE__ __MATX_HOST__ __MATX_DEVICE__ TypeParam legendre(int n, TypeParam in) {
+
+            TypeParam P[3];
+
+            if (n == 0 ) return 1;
+
+            TypeParam x = in;
+
+            P[1] = 1;
+            P[2] = x;
+
+            for(int i = 2; i <= n; i++) {
+              // compute the ith polynomial
+              TypeParam a = (TypeParam(2)*TypeParam(i) - TypeParam(1)) / TypeParam(i);
+              TypeParam b = (TypeParam(i) - TypeParam(1)) / TypeParam(i);
+
+              // shift for next iteration
+              P[0] = P[1];  // P(n-2)
+              P[1] = P[2];	 // P(n-1)
+              P[2] = a * x * P[1] - b * P[0];	  
+            }
+
+            return P[2];
+          }
+
+      public:
+        using matxop = bool;
+        using scalar_type = typename T1::scalar_type;
+
+        __MATX_INLINE__ std::string str() const { return "legrande(" + in_.str() + ")"; }
+
+        __MATX_INLINE__ LegendreOp(int order, const T1 in, const int axis) : in_(in), order_(order), axis_(axis) {}
```

Comment:
const & on `in`

Reply:
I'm avoiding references for operators now.  it causes issues with rvalue vs lvalue parameters.